core/src/main/scala/kafka/log/LogCleanerManager.scala

```diff
@@ -143,6 +143,7 @@ private[log] class LogCleanerManager(val logDirs: Array[File], val logs: Pool[To
           inProgress.put(topicAndPartition, LogCleaningPaused)
         case Some(state) =>
           state match {
+            case LogCleaningPaused => {}
```

Comment:
@soumyajit-sahu : Thanks for the patch. This is definitely caused by a bug. I am not sure if this is the best place to fix it though. For a given topic partition, abortAndPauseCleaning() is expected to be called from a single ReplicaFetcherThread. That thread will call abortAndPauseCleaning(), do something on the log, and call resumeCleaning(). So, when abortAndPauseCleaning() is called, we don't expect the partition to be in the LogCleaningPaused state. The fact that you are hitting this issue indicates that either the previous call of abortAndPauseCleaning() fails to call resumeCleaning() afterwards or somehow abortAndPauseCleaning() is being called concurrently from another thread. So, we probably should figure out which case triggered this and fix it at the source, instead of hiding the issue here. 

Were there any other errors before the first occurrence of IllegalStateException that you reported? Was the topic being deleted and recreated at that time? Is this issue easily reproducible?